File: Team2/Riya_Gupta/RotateArray.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    
    void reverse(vector<int> &nums , int start , int end)
    {
        while(start < end)
        {
            int temp = nums[start];
            nums[start] = nums[end];
            nums[end] = temp;
            start++;
            end--;
        }
    }
    
    
    void rotate(vector<int> &nums,int key)
    {
        
        
        int length = nums.size();
        key = key%length;                    // to keep rotations less than the key as it behave in a cyclic manner
        
        
        reverse(nums , 0, length-1);           // reverse the whole vector
        
        reverse(nums , 0 , key-1);              // reverse the vector upto index k-1
        
        reverse(nums , key , length-1);         // reverse the remaining vector
    }
 
};

File: Team2/kusumlata/rotatearray.java
class Solution {
    public static void reverse(int[] a, int i,int j) {
        int li=i;
        int ri=j;
        while(li<ri){
            int temp=a[li];
            a[li]=a[ri];
            a[ri]=temp;
            li++;
            ri--;
        }
        
    }
    public static void rotate(int[] a,int k){
        k=k% a.length;
        if(k<0){
            k=k+a.length;   
        }
    reverse(a,0,a.length-k-1);
     reverse(a,a.length-k,a.length-1);
     reverse(a,0,a.length-1);
    }
}
   
